### PR TITLE
Darreldonald savebutton

### PR DESCRIPTION
**Pull request by DarrelDonald**
added a save button to edit detectors page in graduation flow
    saves code in the users working branch
    link opens code in devops  in a new tab

### CODE REVIEW

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 1]
```diff
@@ -186,6 +189,10 @@ export class OnboardingFlowComponent implements OnInit {
   openTimePickerSubject: BehaviorSubject<boolean> = new BehaviorSubject(false);
   failureMessage: string = "";
   PPERedirectTimer: number = 10;
+  Organization: string = "";
```
Comment (rekhaswaminathan) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts:192]: This file is getting too big already and hard to read. Let's not add properties which are related to the config of the resource here. Let's have a separate class and use the class object here.

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html [hunk 2]
```diff
@@ -193,20 +193,20 @@ <h4 class="modal-title">Access Denied</h4>
   </fab-dialog>
 
 
-  <fab-panel [isOpen]="publishSuccess || publishFailed" [isLightDismiss]="true" [type]="PanelType.custom"
+  <fab-panel [isOpen]="publishSuccess || publishFailed || saveSuccess || saveFailed" [isLightDismiss]="true" [type]="PanelType.custom"
```
Comment (rekhaswaminathan) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html:196]: Not sure if am understanding this correctly. Its open under all conditions ?

Comment (DarrelDonald) [reply AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html:196]: yes. all will be false when the panel should be closed

Comment (rekhaswaminathan) [reply AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html:196]: I guess am trying to understand role of`publishFailed` and `saveFailed`. The `<h5>` inside the panel shows a message saying detector was saved. So it should appear if `saveSuccess` or `publishSuccess` is true, correct?

Comment (DarrelDonald) [reply AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html:196]: they are so the panel shows on failure. without them theres no notification that the publishing/saving failed. the `<h5>` has a condition above it for one success flags. there are other messages that show in the panel depending on what flags are true

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 3]
```diff
@@ -186,6 +190,13 @@ export class OnboardingFlowComponent implements OnInit {
   openTimePickerSubject: BehaviorSubject<boolean> = new BehaviorSubject(false);
   failureMessage: string = "";
   PPERedirectTimer: number = 10;
+  DevopsConfig: DevopsConfig = {
+    organization: "",
+    project: "",
+    folderPath: "",
+    repository: ""
+  };
+  FolderPath: string = "";
```
Comment (rekhaswaminathan) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts:199]: Is this used anywhere?

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 4]
```diff
@@ -186,6 +190,13 @@ export class OnboardingFlowComponent implements OnInit {
   openTimePickerSubject: BehaviorSubject<boolean> = new BehaviorSubject(false);
   failureMessage: string = "";
   PPERedirectTimer: number = 10;
+  DevopsConfig: DevopsConfig = {
+    organization: "",
```
Comment (rekhaswaminathan) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts:194]: nit: you can initialize the values in the class, if possible.

**Review by rekhaswaminathan — APPROVED** [review]

**Pull request merged**